src/socket.rs

```diff
@@ -1415,6 +1472,31 @@ impl Socket {
         }
     }
 
+    /// Set the value of the `IP_RECVTTL` option for this socket.
+    ///
+    /// If enabled, the `IP_TTL` ancillary message is passed with
+    /// incoming packets.
+    #[cfg(not(any(
+        target_os = "fuschia",
+        target_os = "illumos",
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "redox",
+        target_os = "solaris",
+    )))]
+    pub fn set_recv_ttl(&self, recv_ttl: bool) -> io::Result<()> {
```

Comment:
Can you move this to another pr, that would be quicker to merge, and please also add a getter, following the same pattern for all the other options.

Reply:
I only introduced this because I wanted some control message to use in tests, I've moved to use `recv_tos` and `IP_TOS` instead so I don't need this.